lib/ansible/modules/crypto/openssh_cert.py

```diff
@@ -0,0 +1,542 @@
+                        str(timeobj.month).zfill(2) +
+                        str(timeobj.day).zfill(2) +
+                        str(timeobj.hour).zfill(2) +
+                        str(timeobj.minute).zfill(2) +
+                        str(timeobj.second).zfill(2)
+                    )
+                else:
+                    validity += "19700101010101"
+
+                validity += ":"
+
+                if self.valid_to == "forever":
+                    # on ssh-keygen versions that have the year 2038 bug this will cause the datetime to be 2038-01-19T04:14:07
+                    timeobj = datetime(MAXYEAR, 12, 31)
+                else:
+                    timeobj = self.convertToDatetime(module, self.valid_to)
+
+                validity += (
+                    str(timeobj.year).zfill(4) +
+                    str(timeobj.month).zfill(2) +
+                    str(timeobj.day).zfill(2) +
+                    str(timeobj.hour).zfill(2) +
+                    str(timeobj.minute).zfill(2) +
+                    str(timeobj.second).zfill(2)
+                )
+
+                args.extend(["-V", validity])
+
+            if self.type == 'host':
+                args.extend(['-h'])
+
+            if self.identifier:
+                args.extend(['-I', self.identifier])
+            else:
+                args.extend(['-I', ""])
+
+            if self.principals:
+                args.extend(['-n', ','.join(self.principals)])
+
+            if self.options:
+                for option in self.options:
+                    args.extend(['-O'])
+                    args.extend([option])
+
+            args.extend(['-P', ''])
+
+            try:
+                temp_directory = tempfile.mkdtemp()
+                copy2(self.public_key, temp_directory)
+                args.extend([temp_directory + "/" + os.path.basename(self.public_key)])
+                module.run_command(args, environ_update=dict(TZ="UTC"))
+                copy2(temp_directory + "/" + os.path.splitext(os.path.basename(self.public_key))[0] + "-cert.pub", self.path)
+                rmtree(temp_directory, ignore_errors=True)
+                proc = module.run_command([self.ssh_keygen, '-L', '-f', self.path])
+                self.cert_info = proc[1].split()
+                self.changed = True
+            except Exception as e:
+                try:
+                    self.remove()
+                    rmtree(temp_directory, ignore_errors=True)
+                except OSError as exc:
+                    if exc.errno != errno.ENOENT:
+                        raise CertificateError(exc)
+                    else:
+                        pass
+                module.fail_json(msg="%s" % to_native(e))
+
+        file_args = module.load_file_common_arguments(module.params)
+        if module.set_fs_attributes_if_different(file_args, False):
+            self.changed = True
+
+    def convertToDatetime(self, module, timestring):
```

Comment:
Out of curiosity: is there a reason why you use camelCase instead of under_scores for method names?

Reply:
Because I am used to it and I personally think it is easier to read. However i know that it's much more common to use snake case  in python and I forgot to rename that one.

Reply:
I tried to use camelCase in Python for some time some years ago for the same reason, but eventually gave up (since all library calls etc. don't use it, and mixing both styles is even worse ;) ) and switched to snake_case. I was just curious what's your motivation ;)